cmd/kubeadm/app/cmd/phases/join/kubelet.go

```diff
@@ -117,8 +140,38 @@ func runKubeletStartJoinPhase(c workflow.RunData) (returnErr error) {
 	}
 	bootstrapKubeConfigFile := filepath.Join(data.KubeConfigDir(), kubeadmconstants.KubeletBootstrapKubeConfigFileName)
 
-	// Deletes the bootstrapKubeConfigFile, so the credential used for TLS bootstrap is removed from disk
-	defer os.Remove(bootstrapKubeConfigFile)
+	// Do not delete the bootstrapKubeConfigFile at the end of this function when
+	// using ControlPlaneKubeletLocalMode. The KubeletWaitBootstrapPhase will delete
+	// it when the feature is enabled.
+	if !features.Enabled(initCfg.FeatureGates, features.ControlPlaneKubeletLocalMode) {
+		// Deletes the bootstrapKubeConfigFile, so the credential used for TLS bootstrap is removed from disk
+		defer func() {
+			if err := os.Remove(bootstrapKubeConfigFile); err != nil {
+				klog.Warning(err)
```

Comment:
If we need to check the error of `os.Remove`, we'd better to check whether the file exists before removing it. Or keep it the way it is and simply ignore the return error, like
```
_ = os.Remove(bootstrapKubeConfigFile)
```

Reply:
i recall an old discussion about this in the lines "cleaning the BS kubeconfig is a best effort".

ok, the old code just ignored the error and did not handle it, so this warning seems out of scope for this PR. @chrischdi we can just remove it in the two places? sorry for the back and forth.

```
_ = os.Remove(bootstrapKubeConfigFile)
```

Reply:
it seems like something we can change based on a FR by a user in the future.

Reply:
no worries. Fixed both to `_ = os.Remove(bootstrapKubeConfigFile)`